community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -167,71 +160,186 @@ public Commitment append( TransactionRepresentation transaction, long expectedTr
                         "Received " + transaction + " with txId:" + expectedTransactionId +
                         " to be applied, but appending it ended up generating an unexpected txId:" + transactionId );
             }
-            return append0( transaction, transactionId );
+            return serializeToTransactionLog( transaction, transactionId );
         }
     }
 
-    private class TransactionCommitment implements Commitment
+    private static class TransactionCommitment implements Commitment
```

Comment:
It's more obvious where the fields come from in the latter case. The two are less coupled, and this makes it easier to convert the inner class to a top level class, if we want to do that later.